lib/prayermate_api.rb

```diff
@@ -40,12 +40,12 @@ def resize_image(url, filename)
       HTTParty.post(api_path("images/resize"), body: { url: url, target_filename: filename }, headers: http_headers)
     end
 
-    def register_with_promo_code(promo_code, first_name, last_name, email, roles)
-      register({ promo_code: promo_code }, first_name, last_name, email, roles)
+    def register_with_promo_code(promo_code, first_name, last_name, email, roles, organisation_details)
+      register({ promo_code: promo_code }.merge(organisation_details), first_name, last_name, email, roles)
```

Comment:
I'm not sure I like this implementation... Because it's called `organisation_details` but it is actually just an arbitrary dictionary. I think either it should explicitly be `register({ promo_code: promo_code }.merge(organisation: organisation_details)` or it should be called `extra_params` or something like that